tests/basics_test.py

```diff
@@ -90,9 +90,9 @@ def test_const(self):
         """Test that modifying const Doodds is not allowed.
         """
         d = challenge.basics.Doodad.get_const()
-        self.assertIsInstance(d, challenge.basics.Doodad)
-        self.assertRaises(TypeError, setattr, d, "name", "k")
-        self.assertRaises(TypeError, setattr, d, "value", 9)
+        self.assertIsInstance(d, challenge.basics.ImmutableDoodad)
+        self.assertRaises(AttributeError, setattr, d, "name", "k")
+        self.assertRaises(AttributeError, setattr, d, "value", 9)
```

Comment:
I don't mind changing the expected exception, but we ought to have some way to make the `isinstance` check work.  Maybe by making both `ImmutableDoodad` and `MutableDoodad` inherit from `Doodad` as an ABC?

That said, I don't know that we want to make that change.  But maybe it'd be good to comment that this situation isn't ideal.